inference-engine/src/gna_plugin/gna2_model_debug_log.cpp

```diff
@@ -158,4 +162,225 @@ void WriteInputAndOutputTextGNAImpl(const Gna2Model & gnaModel, const std::strin
+
+        dumpFile << "\n\t\tBase value for input (B) : " << B << "\n";
+        dumpFile << "\t\tBase value for output (b) : " << segments[k].yBase << "\n";
+        dumpFile << "\t\tSegment slope (S): " << segments[k].Slope << "\n";
+        dumpFile << "\t\tShift (scale) : " << scale << "\n";
+        dumpFile << "\t\ty = ax + b:   a = " << a << ", b = " << b;
+        if (segments[k].Slope != 0) {
+            double x0 = static_cast<double>(B) - ((static_cast<double>(segments[k].yBase) * factor) / segments[k].Slope);
+            dumpFile << ", x0 = " << x0;
+        }
+        dumpFile << "\n";
+    }
+}
+
+static void DumpCompoundBias(std::ostream& dumpFile, const Gna2Tensor& activation) {
+    const Gna2CompoundBias* const bias = static_cast<Gna2CompoundBias*>(activation.Data);
+
+    dumpFile << "\t\tBias: " << bias->Bias << ", multiplier: " << unsigned(bias->Multiplier) << "\n";
+}
+
+void DumpGna2Model(const Gna2Model& gnaModel, const std::string dumpFolderNameGNA, bool dumpData) {
+    std::stringstream dumpFileName;
+    uint32_t opsNo = gnaModel.NumberOfOperations;
+
+    dumpFileName << dumpFolderNameGNA << "Gna2ModelDebugDump_" << opsNo << "_layer" << ".txt";
+
+    std::ofstream dumpFile(dumpFileName.str(), std::ios::out);
+
+    dumpFile << "Layers (operations) count: " << opsNo << "\n";
+    dumpFile << "--------------------------------------------------------------\n\n";
+
+    for (uint32_t i = 0; i < opsNo; i++) {
+        const auto& operation = gnaModel.Operations[i];
+
+        dumpFile << "Layer (operation): " << i << "\n";
+        dumpFile << "Layer (operation) type: " << GetLayerType(operation.Type) << "\n";
+        dumpFile << "Number of possible operands: " << operation.NumberOfOperands << "\n";
+
+        for (uint32_t j = 0; j < operation.NumberOfOperands; j++) {
+            if (operation.Operands[j] == nullptr) {
+                dumpFile << "\tOperand " << j << " == nullptr\n\n";
+                continue;
+            }
+            const auto& operand = *operation.Operands[j];
+            dumpFile << "\tOperand " << j << " (" << GetOperandNames(operation.Type)[j] << ")"
+                << " type: " << GetOperandType(operand.Type) <<
+                " shape: " << GetSimpleString(operand.Shape) << "\n";
+
+            if (operand.Type == Gna2DataTypePwlSegment) {
+                DumpPwl(dumpFile, operand);
+            } else if (operand.Type == Gna2DataTypeCompoundBias) {
+                DumpCompoundBias(dumpFile, operand);
+            } else if (dumpData) {
+                std::vector<uint32_t> elementIndex(operand.Shape.NumberOfDimensions);
+                uint32_t k = 0;
+
+                // NOTE: data is dumped sequentially in lines of 64 values for better screen space usage
+                do {
+                    int32_t value = GetValue(operand, elementIndex);
+                    dumpFile << std::setw(8) << value << (((++k % 64) == 0) ? "\n" : "");
```

Comment:
IMO `8` and `64` should depend on precission,

Maybe a simple helper class for generic Gna2Tensor.Data dumping

Reply:
Moved data dump to a different file, with column dump to align with dnn dump style and allow for easier copying.